File: backend/src/main/java/com/dhairya/Placement_management_system/jobpost/JobPost.java
package com.dhairya.Placement_management_system.jobpost;

import com.dhairya.Placement_management_system.drive.Drive;
import com.dhairya.Placement_management_system.user.User;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.UUID;

@Entity
@Table(name = "job_posts")
@Getter
@Setter
@NoArgsConstructor
public class JobPost {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "drive_id", nullable = false)
    private Drive drive;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "recruiter_id", nullable = false)
    private User recruiter;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String description;

    private String location;

    @Column(name = "salary_range")
    private String salaryRange;

    @Column(nullable = false)
    private Integer vacancies = 1;

    @Column(nullable = false)
    private String status = "OPEN";
}

File: frontend/src/types/jobPost.ts
import type { DriveResponse } from './drive'
import type { UserResponse } from './user'

export interface CreateJobPostRequest {
  driveId: string
  title: string
  description: string
  location?: string
  salaryRange?: string
  vacancies: number
}

export interface UpdateJobPostRequest {
  title?: string
  description?: string
  location?: string
  salaryRange?: string
  vacancies?: number
}

export interface JobPostResponse {
  id: string
  drive: DriveResponse
  recruiter: UserResponse
  title: string
  description: string
  location: string | null
  salaryRange: string | null
  vacancies: number
  status: 'OPEN' | 'FILLED' | 'CANCELLED'
}
